### PR TITLE
End reason

### PR DESCRIPTION
**Pull request by hiruna72**

### CODE REVIEW

File: src/read_fast5.c [hunk 1]
```diff
@@ -258,11 +259,18 @@ herr_t fast5_attribute_itr (hid_t loc_id, const char *name, const H5A_info_t  *i
...
     else if(strcmp("end_reason",name)==0 && H5Tclass==H5T_ENUM){
         flag_new_group_or_new_attribute_read_group = 0;
-//        operator_data->slow5_record->end_reason = value.attr_uint8_t;
-        std::string key = "ns_" + std::string(name); //not stored
-        char warn_message[300];
-        sprintf(warn_message,"Not stored: Attribute %s/%s is not stored yet until we confirm from ONT about its datatype", operator_data->group_name,name);
-        search_and_warn(operator_data,key,warn_message);
+        if(*(operator_data->flag_lossy)==0){
+            if(*(operator_data->flag_header_is_written)==0){
+                const char *enum_labels[] = {"unknown","partial","mux_change","unblock_mux_change","signal_positive","signal_negative"};
```
Comment (hiruna72) [inline src/read_fast5.c:264]: @sashajenner can you please check if this implementation is okay? This is where I add end_reason to aux_map.

File: src/split.c [hunk 2]
```diff
@@ -119,8 +119,23 @@ void split_child_worker(proc_arg_t args,
                 if(lossy==0){
                     slow5_aux_meta_t* aux_ptr = slow5File_i->header->aux_meta;
                     uint32_t num_aux_attrs = aux_ptr->num;
+                    int aux_add_fail = 0;
                     for(uint32_t r=0; r<num_aux_attrs; r++){
-                        if(slow5_aux_meta_add(slow5File->header->aux_meta, aux_ptr->attrs[r], aux_ptr->types[r])){
+                        if(aux_ptr->types[r]==SLOW5_ENUM || aux_ptr->types[r]==SLOW5_ENUM_ARRAY){
```
Comment (hiruna72) [inline src/split.c:124]: @sashajenner also this is how an aux_map is copied to a new slow5 file. Here, I create many small slow5 files from a big one.

File: src/s2f.c [hunk 3]
```diff
@@ -61,17 +64,41 @@ void set_hdf5_attributes(hid_t group_id, group_flags group_flag, slow5_hdr_t *he
             break;
         case RAW:
             // add Raw attributes
-            add_attribute(group_id,"start_time",slow5_aux_get_uint64(slow5_record, "start_time", &err),H5T_STD_U64LE);
+            if(header->aux_meta){
+                uint64_t start_time = slow5_aux_get_uint64(slow5_record, "start_time", &err);
+                if(err >= 0){
```
Comment (sashajenner) [inline src/s2f.c:69]: err == 0 on success (not necessary to check >=)

File: src/split.c (hunk 2 above)
Comment (sashajenner) [reply src/split.c:124]: looks fine

File: src/read_fast5.c (hunk 1 above)
Comment (sashajenner) [reply src/read_fast5.c:264]: This is fine. Just a bit hard coded that's all (is it possible to set enum_labels dynamically based on the fast5 file?).

Comment (hasindu2008) [reply src/read_fast5.c:264]: will this catch if nanopore changes their enum keys?

Comment (sashajenner) [reply src/read_fast5.c:264]: no lol line 264 is hardcoded

**Pull request merged**